File: frontend/src/entities/QuizQuestion.ts
export type QuizQuestion = {
    id: string;
    questionText: string;
    questionType: "artist" | "track";
    correctAnswer: string;
    correctIndex: number;
    options: string[];
    previewUrl?: string;
    coverUrl?: string;
    trackTitle: string;
    artist: string;
};

File: Backend/Domain/Entities/QuizQuestion.cs
namespace Backend.Domain.Entities;

public class QuizQuestion
{
    public Guid Id { get; set; }
    public string QuestionText { get; set; }
    public string CorrectAnswer { get; set; }
    public List<string> Options { get; set; } = [];
    public int CorrectIndex { get; set; }
    public string QuestionType { get; set; }
    public string PreviewUrl { get; set; }
    public string CoverUrl { get; set; }
    
    public Guid GameSessionId { get; set; }
    public GameSession GameSession { get; set; }
}